src/makeInfinityAdder.js

```diff
@@ -4,7 +4,21 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+  const innerFunction = (a) => {
+    const x = sum;
+
+    if (a !== undefined) {
+      sum += a;
+
+      return innerFunction;
+    }
+    sum = 0;
+
+    return x;
```

Comment:
Returning `x` here is incorrect because it holds the value of `sum` before the addition. You should return the updated `sum` instead, and then reset `sum` to 0 after returning its value.